src/main/java/io/github/mishkun/ideavimsneak/IdeaVimSneakExtension.kt

```diff
@@ -124,9 +125,18 @@ class IdeaVimSneakExtension : VimExtension {
     private enum class Direction(val offset: Int) {
         FORWARD(1) {
             override fun findBiChar(charSequence: CharSequence, position: Int, charone: Char, chartwo: Char): Int? {
+                var match: Boolean
+
                 for (i in (position + offset) until charSequence.length - 1) {
-                    if (charSequence[i] == charone &&
-                            charSequence[i + 1] == chartwo) {
+                    match = charSequence[i].equals(charone, ignoreCase = OptionsManager.ignorecase.isSet) &&
+                            charSequence[i + 1].equals(chartwo, ignoreCase = OptionsManager.ignorecase.isSet)
+                    if (OptionsManager.ignorecase.isSet && OptionsManager.smartcase.isSet) {
+                        if (charone.isUpperCase() || chartwo.isUpperCase()) {
+                            match = charSequence[i].equals(charone, ignoreCase = false) &&
+                                    charSequence[i + 1].equals(chartwo, ignoreCase = false)
+                        }
+                    }
+                    if (match) {
```

Comment:
You can extract this logic of matching into separate helper function `fun matches(charSequence: CharSequence, charPosition: Int, charOne: Char, charTwo: Char)` inside enum class if you want.